### ISSUE TITLE
No such file or directory: 'datasets/BUP_20/BUP_20.yaml'

### ISSUE THREAD

**Issue by NagaPadma** [Agricultural-Robotics-Bonn/pagnerf#2]
![Unbenannt](https://github.com/Agricultural-Robotics-Bonn/pagnerf/assets/117601878/4370ba7f-e4b0-4b2d-8938-52fcea5ddcbd)

I couldnt find BUP_20.yaml file when i extracted the BUP20.tar.gz

**Comment by LvisRoot** [comment]
Hi @NagaPadma,

I usually have symlinks to the dataset files for naming convenience as follows:
```
BUP_20.yaml -> CKA_sweet_pepper_2020_summer.yaml
BUP_20.json -> CKA_sweet_pepper_2020_summer.json
``` 
I pushed a change to fix this so the dataloader automatically tries to find the original dataset filenames

It should work now if you pull the from main.

**Comment by LvisRoot** [comment]
I'm closing this one since the issue should be fixed.

**Comment by NagaPadma** [comment]
I would like to bring another thing to your attention. 
When I use below script, the folder named 'CKA_sweet_pepper_2020_summer' gets downloaded.
`bash scripts/get_bup20.sh`                         # ~70GB
When I use below script, the BUP_20 dataset gets downloaded. Is there any ambiguity here?
`bash scripts/get_bup20_mask2former_detections.sh`  # ~58GB"

**Comment by LvisRoot** [comment]
Hi @NagaPadma, I'm not sure if I understand what's going on here.

Can you send me the folder structure you get after running those 2 scripts?

**Comment by NagaPadma** [comment]
![image](https://github.com/Agricultural-Robotics-Bonn/pagnerf/assets/117601878/5542d6d6-a79b-4719-a79d-b88dddf9a78e)
sorry before i attached a renamed folders.
You now seeing the original folder structure.
BUP_20.json and BUP_20.yaml are manullay created by me.

**Comment by LvisRoot** [comment]
`BUP_20` folder looks good, but what's inside of `CKA_sweet_pepper_2020_summer` Can you share that as well?

**Comment by NagaPadma** [comment]
Please find the above image.

**Comment by LvisRoot** [comment]
This is how the dataset structure should look like:
```
BUP_20
├── 20200924
├── 20201001
├── CKA_sweet_pepper_2020_summer.json
├── CKA_sweet_pepper_2020_summer.yaml
├── preds_mask2former_finetuned
└── README.md
```

I put together a quick fix, pleas pull from branch #4 to test it and let me know if that works for you.

**Comment by NagaPadma** [comment]
![image](https://github.com/Agricultural-Robotics-Bonn/pagnerf/assets/117601878/521afd42-f3f2-40df-8ba2-d71c72be7143)
Please find the above dir error.

**Comment by LvisRoot** [comment]
Dear @NagaPadma, That might be just that you're running the script from the dataset folder or something like that.
I think that's the case for the repeated `datasets/BUP_20` path.

otherwise you might need to change `--dataset-path` and `--load-modes imgs semantics instance` so that it finds the data.

I think you can figure it out at this point.

Let me know if it works.

**Comment by LvisRoot** [comment]
Hi @NagaPadma thanks for closing the issue :+1: ,

Did you try #4 in the end? let me know if it worked so I can merge it?

**Comment by NagaPadma** [comment]
yes yes it worked for me.
yiou can merge it.

Also in .\train.sh there are few erros. I modified locally. 
#!/bin/bash

__NV_PRIME_RENDER_OFFLOAD=1 \
__GLX_VENDOR_LIBRARY_NAME=nvidia \
WISP_HEADLESS=1 \
python main_interactive.py \
--config "configs/bup20/best.yaml" \
--dataset-path "/home/veerla/00_Projects/PAg/pagnerf/datasets/BUP_20" \
--pretrained "" \
--dataset-num-workers 6 \
--log-dir "_results/logs/runs/bup20" \
--dataset-center-idx "10" \
--exp-name "seq_10" \
--save-preds \
--inst-outlier-rejection \
--inst-num-dilations 1 \
--inst-conf-enable \
--sem-conf-enable \
--max-depth 1.2 \
--inst-segment-reg-weight 0.1 \
--inst-segment-reg-epoch-start 101 \
--load-modes imgs semantics instance "/home/veerla/00_Projects/PAg/pagnerf/datasets/CKA_sweet_pepper_2020_summer/preds_mask2former_finetuned" inst_conf sem_conf \
--class-labels bg red yellow green mixed_red mixed_yellow \
--valid-every 200
You can compare and merge it in the project.

**Comment by NagaPadma** [comment]
Thank you @LvisRoot for your prompt support. I appreciate your assistance and kindness. Thank you
